src/test/java/dmytro/hadiuchko/springboot/service/impl/BookServiceImplTest.java

```diff
@@ -0,0 +1,244 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import dmytro.hadiuchko.springboot.dto.book.request.CreateBookRequestDto;
+import dmytro.hadiuchko.springboot.dto.book.responce.BookDto;
+import dmytro.hadiuchko.springboot.dto.book.responce.BookDtoWithoutCategoryIds;
+import dmytro.hadiuchko.springboot.entity.Book;
+import dmytro.hadiuchko.springboot.exception.EntityNotFoundException;
+import dmytro.hadiuchko.springboot.mapper.BookMapper;
+import dmytro.hadiuchko.springboot.repository.BookRepository;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+class BookServiceImplTest {
+    private static final String NEW_AUTHOR = "Harper Lee";
+    private static final String NEW_TITLE = "To Kill a Mockingbird";
+    private static final String NEW_ISBN = "978-3-16-148410-0";
+    private static final BigDecimal NEW_PRICE = BigDecimal.valueOf(10.99);
+    private static final String NEW_DESCRIPTION = "interesting description.";
+    private static final String NEW_COVER_IMAGE = "tokillamockingbird.jpg";
+    private static final Long BOOK_ID = 1L;
+    private static final String BOOK_TITLE = "some title";
+    private static final String UPDATED_BOOK_TITLE = "updated book's title";
+    private static final int PAGE_SIZE = 10;
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @Mock
+    private BookMapper bookMapper;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
+
+    @Test
+    @DisplayName("Save a new book with valid values")
+    void save_validRequestDto_bookSaved() {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setAuthor(NEW_AUTHOR);
+        requestDto.setTitle(NEW_TITLE);
+        requestDto.setIsbn(NEW_ISBN);
+        requestDto.setPrice(NEW_PRICE);
+        requestDto.setDescription(NEW_DESCRIPTION);
+        requestDto.setCoverImage(NEW_COVER_IMAGE);
+
+        BookDto bookDto = new BookDto();
+        bookDto.setAuthor(requestDto.getAuthor());
+        bookDto.setTitle(requestDto.getTitle());
+        bookDto.setIsbn(requestDto.getIsbn());
+        bookDto.setPrice(requestDto.getPrice());
+        bookDto.setDescription(requestDto.getDescription());
+        bookDto.setCoverImage(requestDto.getCoverImage());
```

Comment:
I think we can build those objects in `before each` method to avoid duplicating it over the other methods, also make use of builder or `@Accessors(chain = true)`